cf/data/data.py

```diff
@@ -10956,9 +10956,10 @@ def flip(self, axes=None, inplace=False, i=False):
         if not iaxes:
             return d
 
-        index = [
-            slice(None, None, -1) if i in axes else slice(None) for i in iaxes
```

Comment:
There's something not right about this, which is causing the unit test to fail on the `self.assertTrue((d.array == array).all())` and some other interactive examples I tried to error, and I initially thought there was a typo and `axes` was meant to be `iaxes`, but then realised that the list comprehension then wouldn't really make sense, at least in that it would short circuit to `[slice(None, None, -1) for i in iaxes]`...

Anyway, the following as suggested in the PR code change works to fix the unit test, though I am wrapping my head around the logic used for `flip` and whether this suggestion is applying the intended logic. Please let me know what you think!

Reply:
Ah, or maybe not, since this doesn't seem right:

```pycon
In [4]: import numpy as np
   ...: import cf
   ...: 
   ...: a = np.arange(3 * 4 * 5).reshape(3, 4, 5) + 1.0
   ...: d = cf.Data(a)
   ...: 
   ...: print(d)
   ...: print(d.flip())
[[[1.0, ..., 60.0]]]
[[[41.0, ..., 20.0]]]
```

Some advice would be appreciated here, instead of me trying to guess what's not right...